### PR TITLE
CAT-1337 Update checkstyle to 6.7

### PR DESCRIPTION
**Pull request by robertpainsi**
https://jira.catrob.at/browse/CAT-1337

### CODE REVIEW

**Comment by robertpainsi** [comment]
There are two ways to update the checkstyle version:
- http://stackoverflow.com/a/26552325
- http://stackoverflow.com/a/26549397

I chose the second one because all the library versions like actionbarsherlock, robotium, pmd, etc are defined in the dependencies block. So there is only one place for updating libraries.

**Comment by robertpainsi** [comment]
Wow, a lot of tests failed. However *'Checkstyle: 0 warnings from one analysis.'*

**Comment by aried3r** [comment]
Admit that you've only done this ticket because it's number 1337. ;)

Any noteworthy changes or new features in 6.6 we should take advantage of?

**Comment by robertpainsi** [comment]
^^ I also should've waited 88 PRs.

> Any noteworthy changes or new features in 6.6 we should take advantage of?

Yeah, EmptyCatchBlock. I just created a jira ticket for that (https://jira.catrob.at/browse/CAT-1374).

**Comment by robertpainsi** [comment]
Hmm, I'm missing the test results, therefore:

**Comment by robertpainsi** [comment]
Hmm, I'm missing the test results, therefore:
https://jenkins.catrob.at/view/All-Categories/view/Catroid-multi-job/job/Catroid-Build-SourceTests-StaticChecks/2269/

**Comment by tobijdc** [comment]
6.7 released
https://github.com/checkstyle/checkstyle/releases/tag/checkstyle-6.7

**Comment by robertpainsi** [comment]
@tobijdc: Thanks for the hint, updated!

**Comment by obusher** [comment]
`12:30:16 * What went wrong:
12:30:16 Execution failed for task ':checkstyle'.
12:30:16 > java.lang.ClassNotFoundException: com.puppycrawl.tools.checkstyle.CheckStyleTask `

**Comment by robertpainsi** [comment]
Yeah, I know. Updating to 6.7 worked, so maybe the library for 6.8 is not available yet...

**My plan**
* ~~Nobody will notice that the update doesn't work until the merge stopped is revoked~~
* Wait until the problem is resolved by itself :wink:

**Comment by tobijdc** [comment]
[6.8.1](https://github.com/checkstyle/checkstyle/releases/tag/checkstyle-6.8.1) =/
(I subscribe via RSS to their release tags)

**Comment by robertpainsi** [comment]
Checkstyle 6.8+ does not work with gradle at the moment. https://discuss.gradle.org/t/checkstyle-plugin-does-not-work-with-cs-6-8/10353
I will keep an eye on [gradle#3314](https://issues.gradle.org/plugins/servlet/mobile#issue/GRADLE-3314)

**Comment by obusher** [comment]
ok to test

**Comment by robertpainsi** [comment]
I'll upgrade checkstyle to 6.7 only.
Gradle would need an upgrade as well and the checkstyle 6.8 bug is still not fixed.

**Comment by obusher** [comment]
ok to test

**Comment by obusher** [comment]
ok to test

**Comment by obusher** [comment]
https://jenkins.catrob.at/view/All-Categories/view/Catroid/job/Catroid-Device-Single-ClassAndPackage-Test/297/ is green!

File: build.gradle [hunk 1]
```diff
@@ -70,6 +70,7 @@ dependencies {
     androidTestCompile 'com.google.dexmaker:dexmaker-mockito:1.2'
     androidTestCompile 'org.mockito:mockito-core:1.10.19'
     pmd 'net.sourceforge.pmd:pmd:5.1.1'
+	checkstyle 'com.puppycrawl.tools:checkstyle:6.7'
```
Comment (avihric) [inline build.gradle]: You missed this one @obusher

Comment (obusher) [inline build.gradle]: LOL!

Comment (robertpainsi) [inline build.gradle]: Screw you whitespaces!!! Will be fixed with the next checkstyle update :wink:

**Comment by robertpainsi** [comment]
JFI: gradle 2.7 fixes the bug with checkstyle 6.8+ https://github.com/gradle/gradle/pull/491

**Comment by obusher** [comment]
:smile_cat:

**Pull request merged**